queue.js

```diff
@@ -0,0 +1,75 @@
+const {
+    SQSClient,
+    ReceiveMessageCommand,
+    DeleteMessageCommand,
+} = require('@aws-sdk/client-sqs');
+const retry = require('retry');
+
+const sqsClient = new SQSClient({
+    region: process.env.AWS_REGION,
+    credentials: {
+        accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+        secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+    },
+});
+
+const queueURL = process.env.SQS_QUEUE_URL;
+
+const getMessage = async () => {
+    return new Promise((resolve, reject) => {
+        const receiveMessageCommand = new ReceiveMessageCommand({
+            QueueUrl: queueURL,
+            MaxNumberOfMessages: 1,
+            WaitTimeSeconds: 2,
+        });
+
+        const operation = retry.operation();
+
+        let res;
+        operation.attempt(async (currentAttempt) => {
+            try {
+                res = await sqsClient.send(receiveMessageCommand);
+                resolve(res.Messages ? res.Messages[0] : null);
+            } catch (err) {
+                if (operation.retry(err)) {
+                    return;
+                }
+            }
+            reject(operation.mainError());
```

Comment:
when will this line be executed?

Reply:
Good point. I'm going to address that as I've described [here](https://github.com/alex-808/flashcard-service/pull/1#discussion_r1300892308) so the error gets a chance to propagate out eventually